runtime/lua/vim/lsp/buf.lua

```diff
@@ -143,6 +143,63 @@ local function select_client(method, on_choice)
   end
 end
 
+--- Formats the current buffer.
+---@param opts table
+---     formatting_options: table: Can be used to specify FormattingOptions.
+---                                Some unspecified options will be automatically derived from the current
+---                                Neovim options.
+---                                @see https://microsoft.github.io/language-server-protocol/specification#textDocument_formatting
+---     timeout_ms: string
+---     filter: function
+---     ids: table
+---     names: table
+---     id: string
+---     name: string
+
+function M.format(opts)
+  local bufnr = opts.bufnr or vim.api.nvim_get_current_buf()
+  local clients = vim.lsp.buf_get_clients()
```

Comment:
Just in case the bufnr opt makes it through:

```suggestion
  local clients = vim.lsp.buf_get_clients(bufnr)
```

Reply:
that line is redundant